system/handlers/core/AssetDownload.cfc

```diff
@@ -178,54 +184,55 @@ component {
 		return ReReplace( arguments.assetTitle, "\.#arguments.extension#$", "" ) & "." & arguments.extension;
 	}
 
-	private void function _checkDownloadPermissions( event, rc, prc ) output=false {
+	private boolean function _checkDownloadPermissions( event, rc, prc ) output=false {
```

Comment:
Is this the best name for the method if it's now returning a boolean? Doesn't really explain what the boolean signifies...

Reply:
Yeah, is serving two purposes which is a little funky. i.e. it _does_ perform a check in that it will block the user. But having it also return a value. Bit weak as a pattern.

Reply:
Have split things out into separate methods now.